test/e2e/destroypod_test.go

```diff
@@ -48,13 +49,30 @@ const (
 	timeoutRequestDuration = 35 * time.Second
 )
 
+// testToDestroy for table-driven testing.
+var testToDestroy = []string{
+	// Destroy pods which is receiving the requests.
+	"pods",
+	// Destroy activator pods.
+	"activators",
+}
+
 func TestDestroyPodInflight(t *testing.T) {
-	t.Parallel()
+	// Not running in parallel as this test delete activator pods
 	cancel := logstream.Start(t)
 	defer cancel()
-
 	clients := Setup(t)
 
+	for _, tc := range testToDestroy {
+		t.Run(tc, func(t *testing.T) {
+			cancel := logstream.Start(t)
+			defer cancel()
+			testDestroyPodInflight(t, clients, tc)
```

Comment:
I'd propose to not pass the testcase name but to pass a removal function of the form `func(clients) error`.

You can then create

`func killActivatorPods(clients) error` and `func killRevisionPods(revisionName) func(clients) error` and reuse them for both tests if you choose to act on the comments I left below.